examples/maven/maven.yaml

```diff
@@ -1,5 +1,72 @@
 ---
 apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: cosign-upload-blob
+spec:
+  workspaces:
+    - name: cosign-ws
+      description: workspace
+  params:
+    - name: image
+      description: The image URI where the artifact will be uploaded
+      default: ttl.sh/bradbeck/cosignblob:latest
+    - name: artifact
+      description: Relative path in the workspace of the artifact to be uploaded
+  results:
+    - name: IMAGE_URL
+      description: image url
+    - name: IMAGE_DIGEST
+      description: image digest
+  steps:
+    - name: upload
+      image: gcr.io/projectsigstore/cosign:v1.8.0@sha256:12b4d428529654c95a7550a936cbb5c6fe93a046ea7454676cb6fb0ce566d78c
+      workingDir: $(workspaces.cosign-ws.path)
+      args:
+        - upload
+        - blob
+        - -f
+        - $(params.artifact)
+        - --output-file
+        - out.txt
+        - $(params.image)
+    - name: url-digest
+      image: docker.io/library/bash
+      workingDir: $(workspaces.cosign-ws.path)
+      script: |
+        echo -n "$(params.image)" | tr -d '[:space:]' | tee $(results.IMAGE_URL.path)
+        cat out.txt | sed -e 's/.*@//' | tr -d '[:space:]' | tee $(results.IMAGE_DIGEST.path)
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: cosign-attach-sbom
+spec:
+  workspaces:
+    - name: cosign-ws
+      description: workspace
+  params:
+    - name: image
+      description: The image URI to which the SBOM will be attached
+    - name: sbom
+      description: Relative path in the workspace of the SBOM to be attached
+    - name: type
+      description: The type of SBOM (spdx|cyclonedx|syft)
+      default: spdx
+  steps:
+    - name: attach
+      image: gcr.io/projectsigstore/cosign:v1.8.0@sha256:12b4d428529654c95a7550a936cbb5c6fe93a046ea7454676cb6fb0ce566d78c
+      workingDir: $(workspaces.cosign-ws.path)
+      args:
+        - attach
```

Comment:
This is where it would be nice to have https://github.com/tektoncd/community/blob/main/teps/0011-redirecting-step-output-streams.md so we can capture the digest of the uploaded SBOM and then emit result like `SBOM_IMAGE_URL` & `SBOM_IMAGE_DIGEST` and then let chains do the signing.

Reply:
Yup, definitely not signed. regctl can attach an artifact and emit the resulting digest, but it uses a slightly different experimental tagging scheme since OCI is still sorting this out in the working group. Generating that tag would also be easy to do, and we can either push to that tag with another tool, or get the digest of that tag leaving cosign to do the push.